app/assets/javascripts/deploys.js

```diff
@@ -88,6 +88,18 @@ function update_tasks(progress_data) {
   });
 }
 
+function update_progress(data, finished) {
+  if ("progress" in data) {
+    if ("total_progress" in data.progress) {
+      update_progress_bar(
+          data.progress.total_progress, data.error, finished)
+    }
+    if ("tasks_progress" in data.progress) {
+      update_tasks(data.progress.tasks_progress, finished)
+    }
+  }
```

Comment:
Early returns for the win ;)
```suggestion
  if (!("progress" in data)) {
      return
  }
  if ("total_progress" in data.progress) {
      update_progress_bar(data.progress.total_progress, data.error, finished)
  }
  if ("tasks_progress" in data.progress) {
      update_tasks(data.progress.tasks_progress, finished)
  }
```

Reply:
Thanks! Changed in https://github.com/SUSE/blue-horizon/pull/24/commits/cd341b4e7fbc3670796a91c18b743238fd3ae4f5